crates/subspace-networking/src/node_runner.rs

```diff
@@ -67,10 +75,22 @@ impl NodeRunner {
             query_id_receivers: HashMap::default(),
             next_subscription_id: 0,
             topic_subscription_senders: HashMap::default(),
+            pieces_by_range_handler,
         }
     }
 
     pub async fn run(&mut self) {
+        // Retrieve the pieces-by-range handler object.
+        let pieces_by_range_handler = self
+            .pieces_by_range_handler
+            .take()
+            .expect("Pieces-by-range handler must be set at this point.");
```

Comment:
If we call `run` twice, it will panic, right? We can take `self` here by value and it should do

Reply:
I mean we can take ownership of self:
```rust
     pub async fn run(mut self) {
```